fundamentals/fundamentals-bio/src/main/scala/izumi/functional/bio/data/FreeInterpreter.scala

```diff
@@ -0,0 +1,20 @@
+package izumi.functional.bio.data
+
+/**
+  * Should executed scenario steps in a single context and save all scenario metrics.
+  */
+trait FreeInterpreter[F[+_, +_], S[_[+_, +_], _, _], Scope] {
+  /** Single step execution. */
+  def interpret[E, A](scope: Scope, s: S[F, E, A]): F[Nothing, FreeInterpreter.Result[F, S, Scope]]
```

Comment:
The scope parameter may be implemented on the level of interpreter for `F` and removed from the base definition?